Gem/Code/Source/Performance/100KEntityExampleComponent.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <Performance/HighInstanceExampleComponent.h>
+
+namespace AtomSampleViewer
+{
+    /*
+        This test loads a 46x46x46 lattice of entities with randomized meshes and materials. This test is used as a simple cpu performance stress test for Atom.
+
+        The assets that are applied to the entities are chosen from the 
+        "allow-list" of models and materials.
+    */
+    class HundredKEntityExampleComponent
```

Comment:
I guess it would be better to rename this to a number-unrelated name. What if later we want to adjust the example to be 200k or 50k? And searching becomes difficult because here already has "HundredK" and "100K".

Reply:
What would you suggest as a name? I went with numbers because it's how we have described our cpu performance goal, 100K draws with 10K entities.

Reply:
What do you think of directly CpuPerfomanceGoalExampleComponent?
Or I am thinking of defining a performance level (similar to what we did in screenshot comparison), and call this PerformanceLevelOneExampleComponent. So later if we want to raise the bar, we can add a level 2 and adjust the numbers in level 1. Maybe others have better idea.

Reply:
I'm actually okay with having the numbers in the name for now. It's very clear and it's a 2-way-door decision. BUT another approach I thought of is to make a generic "HighEntityExampleComponent" sample, instead of having separate 100KEntity and 100KDraw10KEntity samples. The lattice component provides controls for adjusting the scale, we could add controls for adjusting the lights, and then also add some "Preset" buttons to the sidebar. Then when you open the same you can adjust all the settings as you want, or you can click one of the preset buttons to pick the "100KEntity" and "100KDraw10KEntity" presets, or whatever other presets we want to add in the future.

Reply:
I also prefer the numbers because it's more informative to people outside Atom or even Amazon. They can understand what is being tested faster.